spec/chewy/strategy/delayed_sidekiq_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'spec_helper'
+
+if defined?(Sidekiq)
+  require 'sidekiq/testing'
+  require 'mock_redis'
+
+  describe Chewy::Strategy::DelayedSidekiq do
+    around do |example|
+      Chewy.strategy(:bypass) { example.run }
+    end
+
+    before do
+      redis = MockRedis.new
+      allow(Sidekiq).to receive(:redis).and_yield(redis)
+      Sidekiq::Worker.clear_all
+    end
+
+    before do
+      stub_model(:city) do
+        update_index('cities') { self }
+      end
+
+      stub_index(:cities) do
+        index_scope City
+      end
+    end
+
+    let(:city) { City.create!(name: 'hello') }
+    let(:other_city) { City.create!(name: 'world') }
+
+    specify do
+      expect { [city, other_city].map(&:save!) }
+        .not_to update_index(CitiesIndex, strategy: :delayed_sidekiq)
+    end
+
+    specify do
+      expect(Sidekiq::Client).to receive(:push).with(
+        hash_including(
+          'queue' => 'chewy',
+          'at' => an_instance_of(Integer),
+          'class' => Chewy::Strategy::DelayedSidekiq::Worker,
+          'args' => ['CitiesIndex', an_instance_of(Integer)]
+        )
+      ).and_call_original
+      Sidekiq::Testing.inline! do
+        expect { [city, other_city].map(&:save!) }
+          .to update_index(CitiesIndex, strategy: :delayed_sidekiq)
+          .and_reindex(city, other_city).only
+      end
+    end
+
+    specify do
+      CitiesIndex.delayed_sidekiq_options({reindex_wrapper: ->(&reindex) { reindex.call }, margin: 1, latency: 3})
+      expect(Sidekiq::Client).to receive(:push).with(
+        hash_including(
+          'queue' => 'chewy',
+          'at' => an_instance_of(Integer),
+          'class' => Chewy::Strategy::DelayedSidekiq::Worker,
+          'args' => ['CitiesIndex', an_instance_of(Integer)]
+        )
+      ).and_call_original
+
+      Sidekiq::Testing.inline! do
+        expect { [city, other_city].map(&:save!) }
+          .to update_index(CitiesIndex, strategy: :delayed_sidekiq)
+          .and_reindex(city, other_city).only
+      end
+    end
+
+    specify do
```

Comment:
Could you add meaningful names to the test cases? I know we have  anonymous test in the chewy test suite, but it'd be better to avoid them unless the case is obvious.

Reply:
makes sense. I'll do!